### PR TITLE
Add FromString for hexadecimal string values and use error instead panic

### PR DESCRIPTION
**Pull request by lonewolf3739**
Hey @lukechampine 
First of all, thanks for publishing the `uint128` package. I found there is no func for creating `uint128` from hexadecimal values, so I created this PR. And also returning multiple values for error recovery seems better choice (we don't have nested error anywhere). can you please review and accept this PR?

### CODE REVIEW

**Comment by lukechampine** [comment]
Thanks for the contribution!

My preference is to `panic` on bad input. This makes it easier to chain operations. We already panic on division-by-zero (`math/big` also does this). Clients who need to prevent panics can validate the input themselves, just like they must prevent divide-by-zero errors themselves. I'm also not keen on changing the existing function signatures unless there's a compelling reason to do so.

However, it's pretty rare for string-parsing APIs to panic. `strconv.Atoi` returns an error, `(big.Int).SetString` returns a bool, etc. This makes sense, because it's a bit unreasonable for the caller to validate these inputs themselves. So I'm okay with `FromString` returning an error.

Finally, I would expect `FromString` to accept decimal, not hexadecimal. Perhaps we should accept multiple bases (2, 10, and 16), like `SetString` does.

To summarize, I think we should make `FromString` accept decimal (or multiple bases), and leave everything else as-is. `FromString` can return an error, but let's stick with stdlib errors instead of adding a dependency on `pkg/errors`.

**Comment by lonewolf3739** [comment]
Yes, `panic` on bad input makes sense. Accepting multiple bases(2, 10 and 16) is good idea. Closing this PR. Will create new one. Thanks.

**Comment by lukechampine** [comment]
btw, I think a relatively easy way to implement the parsing would be to call `strconv.ParseUint` repeatedly on each section of the string, consuming as many digits as you can with each pass. For example, to parse base 10, do:
```go
func FromString(s string) (Uint128, error) {
	var res Uint128
	for len(s) > 0 {
		digits := 19 // log2(2^64) = 19
		if len(s) < 19 {
			digits = len(s)
		}
		u, err := strconv.ParseUint(s[:digits], 10, 64)
		if err != nil {
			return Zero, err
		}
		res = res.Mul64(uint64(math.Pow10(digits))).Add64(u)
		s = s[digits:]
	}
	return res, nil
}
```
